### PR TITLE
Add context menu

### PR DESCRIPTION
**Pull request by maxichrome**
Some styling things to work out still. Check out the spacedrive-web deploy and right-click on a file or folder icon!

- Adds a functional context menu
- Adds a `WithContextMenu` wrapper that automatically handles menu state for you
- Implements a basic, not fully-functional, context menu on file icons as a UI demo
- Parameterized the `vite.config.js` and `App.tsx` files and removed the `removeThisSoon` prop
- Added a light/dark theme Tailwind toggle within Storybook

A screenshot of the dark-themed context menu, with three sections and a highlighted hover state on the second-to-last item of the first section called 'Share'

### CODE REVIEW

**Comment by maxichrome** [comment]
A screenshot of the context menu described in the PR's first comment, but with a nested item that currently shows a focused state

File: packages/interface/src/components/file/FileItem.tsx [hunk 1]
```diff
@@ -26,78 +28,128 @@ export default function FileItem(props: Props) {
...
+							<svg
+								width="22"
+								height="22"
+								className="absolute top-1 -right-[1px] z-10 fill-gray-50 dark:fill-gray-500 pointer-events-none"
+								viewBox="0 0 41 41"
+							>
+								<path d="M41.4116 40.5577H11.234C5.02962 40.5577 0 35.5281 0 29.3238V0L41.4116 40.5577Z" />
+							</svg>
+							<div className="absolute flex flex-col items-center justify-center w-full h-full">
+								{/* @ts-ignore */}
```
Comment (Looskie) [inline packages/interface/src/components/file/FileItem.tsx:121]: ts i9gnore :(

File: packages/interface/src/components/file/FileItem.tsx [hunk 2]
```diff
@@ -26,78 +28,128 @@ export default function FileItem(props: Props) {
...
+								{props.iconName && icons[props.iconName] ? (
+									(() => {
+										// @ts-ignore
```
Comment (Looskie) [inline packages/interface/src/components/file/FileItem.tsx:124]: ts-ignore

Comment (maxichrome) [reply packages/interface/src/components/file/FileItem.tsx:124]: I didn't add this... maybe should backtrace the blame and harass whoever added it?

File: packages/interface/src/components/file/FileItem.tsx (hunk 1 above)
Comment (maxichrome) [reply packages/interface/src/components/file/FileItem.tsx:121]: I did not add this

**Review by jamiepine — APPROVED** [review]

**Review by Looskie — APPROVED** [review]
gltm
IMG_1606

**Pull request merged**